ocrd_webapi/routers/user.py

```diff
@@ -0,0 +1,64 @@
+"""
+module for implementing the authentication section of the api
+"""
+import logging
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+
+from ocrd_webapi.authentication import authenticate_user, register_user
+
+router = APIRouter(
+    tags=["User"],
+)
+
+logger = logging.getLogger(__name__)
+# TODO: This may not be ideal, discussion needed
+security = HTTPBasic()
+
+
+@router.get("/user/login")
+async def user_login(auth: HTTPBasicCredentials = Depends(security)):
+    email = auth.username
+    password = auth.password
+    if not (email and password):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            headers={"WWW-Authenticate": "Basic"},
+            detail="Missing e-mail or password field"
+        )
+
+    # Authenticate user e-mail and password
+    try:
+        await authenticate_user(
+            email=email,
+            password=password
+        )
+    except ValueError as error:
+        logger.info(f"User failed to authenticate: {email}, reason: {error}")
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            headers={"WWW-Authenticate": "Basic"},
+            detail=f"{error}"
```

Comment:
The detail of the authentication error should not be returned to users for security reason. Simply "Incorrect username or password" is enough.

Reply:
The problem with the suggested message is that even when the user provides the correct credentials but the account was not approved by an admin it will cause confusion. The only differentiation we need is between these two cases.

Agree. I will change that and return just a message and not the entire exception trace. I can also combine both cases in a single message.